File: MODULO3/UML/DiagramaUML/src/asociacion_tipo_muchos_a_muchos_estudiante_curso/Curso.java
package asociacion_tipo_muchos_a_muchos_estudiante_curso;

import java.util.ArrayList;
import java.util.List;

public class Curso {

    // VARIABLES DE INSTANCIA
    private String idCurso;
    private String nombre;
    private List<Inscripcion> inscripciones;

    // CONSTRUCTORES
    public Curso() {
        this.idCurso = "";
        this.nombre = "";
        this.inscripciones = null;
    }

    public Curso(String idCurso, String nombre) {
        this.idCurso = idCurso;
        this.nombre = nombre;
        this.inscripciones = new ArrayList<>();
    }

    // METODOS SET Y GET
    public String getIdCurso() {
        return idCurso;
    }

    public void setIdCurso(String idCurso) {
        this.idCurso = idCurso;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    // METODO TOSTRING
    @Override
    public String toString() {
        return "Curso{" + "idCurso=" + idCurso + ", nombre=" + nombre + '}';
    }

    // METODO RELACION INSCRIPCION
    public void agregarInscripcion(Inscripcion inscripcion) {
        this.inscripciones.add(inscripcion);
    }
}

File: MODULO3/PYTHON/TRABAJANDO_PYTHON/asociacion_muchos_a_muchos_estudiante_curso/curso.py
from inscripcion import Inscripcion

class Curso:
    def __init__(self, id_curso = "", nombre = ""):
        self._id_curso = id_curso
        self._nombre = nombre
        self.inscripciones = []

    @property
    def id_curso(self):
        return self._id_curso
    @id_curso.setter
    def id_curso(self, value):
        self._id_curso = value
    @property
    def nombre(self):
        return self._nombre
    @nombre.setter
    def nombre(self, value):
        self._nombre = value

    def __str__(self):
        return f"Curso ID: {self.id_curso}, Nombre: {self.nombre}"  

    def agregar_inscripcion(self, inscripcion):
        self.inscripciones.append(inscripcion)


